ckan/views/dataset.py

```diff
@@ -0,0 +1,1175 @@
+
+    def post(self, package_type, id):
+        if u'cancel' in request.form:
+            return h.redirect_to(u'dataset.edit', id=id)
+        context = self._prepare()
+        try:
+            get_action(u'package_delete')(context, {u'id': id})
+        except NotFound:
+            return base.abort(404, _(u'Dataset not found'))
+        except NotAuthorized:
+            return base.abort(
+                403,
+                _(u'Unauthorized to delete package %s') % u''
+            )
+
+        h.flash_notice(_(u'Dataset has been deleted.'))
+        return h.redirect_to(package_type + u'.search')
+
+    def get(self, package_type, id):
+        context = self._prepare()
+        try:
+            pkg_dict = get_action(u'package_show')(context, {u'id': id})
+        except NotFound:
+            return base.abort(404, _(u'Dataset not found'))
+        except NotAuthorized:
+            return base.abort(
+                403,
+                _(u'Unauthorized to delete package %s') % u''
+            )
+
+        dataset_type = pkg_dict[u'type'] or package_type
+
+        return base.render(
+            u'package/confirm_delete.html', {
+                u'pkg_dict': pkg_dict,
+                u'dataset_type': dataset_type
+            }
+        )
+
+
+def follow(package_type, id):
+    """Start following this dataset.
+    """
+    context = {
+        u'model': model,
+        u'session': model.Session,
+        u'user': g.user,
+        u'auth_user_obj': g.userobj
+    }
+    data_dict = {u'id': id}
+    try:
+        get_action(u'follow_dataset')(context, data_dict)
+        package_dict = get_action(u'package_show')(context, data_dict)
+    except ValidationError as e:
+        error_message = (e.message or e.error_summary or e.error_dict)
+        h.flash_error(error_message)
+    except NotAuthorized as e:
+        h.flash_error(e.message)
+    else:
+        h.flash_success(
+            _(u"You are now following {0}").format(package_dict[u'title'])
+        )
+
+    return h.redirect_to(u'dataset.read', id=id)
+
+
+def unfollow(package_type, id):
+    """Stop following this dataset.
+    """
+    context = {
+        u'model': model,
+        u'session': model.Session,
+        u'user': g.user,
+        u'auth_user_obj': g.userobj
+    }
+    data_dict = {u'id': id}
+    try:
+        get_action(u'unfollow_dataset')(context, data_dict)
+        package_dict = get_action(u'package_show')(context, data_dict)
+    except ValidationError as e:
+        error_message = (e.message or e.error_summary or e.error_dict)
+        h.flash_error(error_message)
+    except (NotFound, NotAuthorized) as e:
+        error_message = e.message
+        h.flash_error(error_message)
+    else:
+        h.flash_success(
+            _(u"You are no longer following {0}").format(
+                package_dict[u'title']
+            )
+        )
+
+    return h.redirect_to(u'dataset.read', id=id)
+
+
+# FIXME: Do we need this?
```

Comment:
There is a url registered and a page being rendered currently (although not linked from the UI) so let's keep it. We can check if it works as a separate issue.